docs/advanced_topics_checkpointing.rst

```diff
@@ -0,0 +1,89 @@
+Checkpointing
+=============
+
+.. currentmodule:: nvidia.dali
+
+Checkpointing is a feature in DALI which allows you to save current state of the pipeline to a file.
+Then, you can restore the pipeline from a saved checkpoint and the new pipeline will produce exactly the same outputs as the old one would.
+It is particularly useful for long-running training jobs which are likely to be interrupted.
+
+A checkpoint of DALI pipeline contains information about states of all random number generators used in the pipeline and about the progress of each reader.
+
+Checkpointing API
+-----------------
+
+Enabling checkpointing
+~~~~~~~~~~~~~~~~~~~~~~
+
+To enable checkpointing, set ``enable_checkpointing=True`` when creating a pipeline.
+With this option enabled, DALI will track state of each operator, allowing you to save it on demand.
+Enabling checkpointing shouldn't have any impact on the performance.
+
+.. code-block:: python
+
+  @pipeline_def(..., enable_checkpointing=True)
+  def pipeline():
+      ...
+
+  p = pipeline()
+  p.build()
+
+
+.. note::
+    Readers with ``shuffle_after_epoch=True`` might produce different (but still equally random) output if checkpointing is enabled.
+
+
+Saving a checkpoint
+~~~~~~~~~~~~~~~~~~~
+
+To save a checkpoint, you need to call :meth:`Pipeline.checkpoint` method, which will return a serialized checkpoint as a string.
+Optionally, you can pass filename as an argument and DALI will save the checkpoint there.
+
+.. code-block:: python
+
+  for _ in range(iters):
+      output = p.run()
+
+  # Write the checkpoint to file:
+  checkpoint = p.checkpoint()
+  open('checkpoint_file.cpt', 'wb')
+
+  # Or simply:
+  checkpoint = p.checkpoint('checkpoint_file.cpt')
+
+.. note::
+    Calling :meth:`Pipeline.checkpoint method` may introduce an observable overhead.
```

Comment:
![image](https://github.com/NVIDIA/DALI/assets/44801854/9e3bbf9a-c695-42fd-833a-73d92f291b5a)

I think this does not look like intended.

Reply:
Fixed, thank you